### PR TITLE
Testing tools and sample project included in builds

### PR DESCRIPTION
**Pull request by jbaxleyiii**
This adds `react-apollo/test-utils` to make it easy to unit test your app! Docs to come!

TODO:
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass
- [x] Update CHANGELOG.md with your change

### CODE REVIEW

**Comment by stubailo** [comment]
What does this mean???

**Comment by jbaxleyiii** [comment]
@stubailo this is a WIP to add a number of things!

1. example apps for people contributing to this project to test / play around / replicate issues (base react, react + flow, typescript)
2. react-apollo custom testing utilities with examples! i.e. how do I test my query, HOC, and wrapped component in practice with mocked data
3. flow definitions for people using flow instead of typescript (I'll be adding these to AC once the API updates are done ;) )
4. Logging hooks which will be used by chrome / safari extensions and react-storybook addons to give a deep insight into what is going on inside your app
5. Inline documentation which typescript / flow will be able to read with editor support

**Comment by jbaxleyiii** [comment]
Initial additions are testing utilities and a sample project showing test structure

**Comment by stubailo** [comment]
🎉

File: package.json [hunk 1]
```diff
@@ -111,6 +114,8 @@
   "dependencies": {
     "hoist-non-react-statics": "^1.2.0",
     "invariant": "^2.2.1",
+    "jest": "^16.1.0-alpha.691b0e22",
```
Comment (edvinerikson) [inline package.json]: This just broke our builds
I think it should be moved to dev dependencies

**Comment by jbaxleyiii** [comment]
@edvinerikson sorry about that! New build going out now with it fixed!

**Pull request merged**